pkg/autoscaler/statserver/server.go

```diff
@@ -156,27 +155,40 @@ func (s *Server) Handler(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		// we accept either GOB-encoded or JSON-encoded messages depending on the
+		// we accept either protobuf-encoded or JSON-encoded messages depending on the
 		// message type to ensure safe upgrades.
-		var dec decoder
 		switch messageType {
 		case websocket.BinaryMessage:
-			dec = gob.NewDecoder(bytes.NewBuffer(msg))
+			var wsms metrics.WireStatMessages
+			if err := wsms.Unmarshal(msg); err != nil {
+				s.logger.Errorw("Failed to unmarshal message", zap.Error(err))
```

Comment:
total nit, but:

```suggestion
				s.logger.Errorw("Failed to unmarshal messages", zap.Error(err))
```

Reply:
Strictly, it's a message :)

Reply:
hah, I guess it's a message containing Messages :D

(just seems odd since we log receiving 'message' a few lines down, and there we mean a 'Message' not a 'message'. No big deal though, if you prefer as is it's fine with me)

Reply:
`Failed to unmarshal the object` :)